onebusaway-android/src/main/java/org/onebusaway/android/tripservice/TripService.java

```diff
@@ -111,33 +111,34 @@ public TaskContextImpl(int startId) {
         public void taskComplete() {
             //Log.d(TAG, "Task complete: " + mStartId);
             // If we have notifications, then we can't stop ourselves.
-            if (mNotifications.isEmpty()) {
+            //TODO: Is there actually a reason to not stop ourselves if we are no longer storing notifications?
+//            if (mNotifications.isEmpty()) {
                 stopSelfResult(mStartId);
-            }
+//            }
         }
 
         @Override
         public void setNotification(int id, Notification notification) {
-            mNotifications.put(id, notification);
+//            mNotifications.put(id, notification);
             mNM.notify(id, notification);
         }
 
         @Override
         public void cancelNotification(int id) {
             mNM.cancel(id);
-            mNotifications.remove(id);
+//            mNotifications.remove(id);
             // If there are no more notifications
             // (and nothing else to process in the thread queue?)
             // stop the service.
-            if (mNotifications.isEmpty()) {
+//            if (mNotifications.isEmpty()) {
                 //Log.d(TAG, "Stopping service");
                 stopSelf();
```

Comment:
Same as above comment - I don't think we should stop the service while other Threads may still be executing.